### DISCUSSION TITLE
something wrong happened when i run the example dump file

### DISCUSSION THREAD

**Discussion by pig-pig-man2020** [deepmodeling/reacnetgenerator#1483] [Q&A]
hi, i install reacnetgenerator by "conda install reacnetgenerator -c conda-forge" and something wrong happened with my dump file. so i download the example file you provide from https://bf.njzjz.win/reacnetgenerator/dump.reaxc and the same error still happened.

(base) E:\ChromeCoreDownloads>reacnetgenerator --type lammpsdumpfile -i dump.rea
xc -a C H O
[32m2022-01-08 15:07:37,245[0m - ReacNetGen 1.0.0 - [1;30mINFO:[0m ReacNetGe
nerator: an automatic reaction network generator for reactive
molecular dynamics simulation.

Plase cite: ReacNetGenerator: an automatic reaction network generator
for reactive molecular dynamic simulations, Phys. Chem. Chem. Phys.,
2020, 22 (2): 683–691, doi: 10.1039/C9CP05091D

Jinzhe Zeng (jinzhe.zeng@rutgers.edu), Tong Zhu (tzhu@lps.ecnu.edu.cn)

==================
Features
==================
* Processing of MD trajectory containing atomic coordinates or bond orders
* Hidden Markov Model (HMM) based noise filtering
* Isomers identifying accoarding to SMILES
* Generation of reaction network for visualization using force-directed
  algorithm
* Parallel computing

==================
Simple example
==================
ReacNetGenerator can process any kind of trajectory files containing
atomic coordinates, e.g. a LAMMPS dump file prepared by running “dump 1
all custom 100 dump.reaxc id type x y z” in LAMMPS:
$ reacnetgenerator --dump -i dump.reaxc -a C H O
where C, H, and O are atomic names in the input file. Analysis report
will be generated automatically.

Also, ReacNetGenerator can process files containing bond information,
e.g. LAMMPS bond file:
$ reacnetgenerator -i bonds.reaxc -a C H O

You can running the following script for help:
$ reacnetgenerator -h

[32m2022-01-08 15:07:37,251[0m - ReacNetGen 1.0.0 - [1;30mINFO:[0m Version:
1.0.0  Creation date: 2018-03-11  Update date: 2019-10-07
Read bond information and Detect molecules: 401timestep [00:06, 84.23timestep/s]
Read bond information and Detect molecules: 702timestep [00:06, 167.52timestep/s
Read bond information and Detect molecules: 815timestep [00:07, 180.14timestep/s
Read bond information and Detect molecules: 1001timestep [00:07, 141.78timestep/
s]
Save molecules: 100%|███████████████| 2895/2895 [00:04<00:00, 624
.87molecule/s]
[32m2022-01-08 15:07:50,323[0m - ReacNetGen 1.0.0 - [1;30mINFO:[0m Step 1: D
one! Time consumed (s): 13.070 (Read bond information and Detect molecules)
HMM filter:  10%|██▏                  | 301/2895 [00:04<00:30, 85.68molecule/
HMM filter:  21%|████▏               | 601/2895 [00:05<00:11, 192.96molecul
HMM filter:  31%|██████▏             | 901/2895 [00:05<00:06, 322.87molec
HMM filter:  41%|███████▉           | 1201/2895 [00:05<00:03, 491.64mole
HMM filter:  48%|█████████▏         | 1401/2895 [00:05<00:02, 616.61mo
HMM filter:  59%|███████████▏       | 1701/2895 [00:05<00:01, 831.51
HMM filter:  69%|████████████▍     | 2001/2895 [00:05<00:00, 1012.5
HMM filter:  83%|██████████████▉   | 2401/2895 [00:06<00:00, 1301
HMM filter:  97%|█████████████████▍| 2801/2895 [00:06<00:00, 1
HMM filter: 100%|███████████████████| 2895/2895 [00:06<00:00,
 470.66molecule/s]
[32m2022-01-08 15:07:57,238[0m - ReacNetGen 1.0.0 - [1;30mINFO:[0m Step 2: D
one! Time consumed (s): 6.914 (HMM filter)
Indentify isomers:   1%|▏                | 1/102 [00:04<07:50,  4.66s/molecule]
Indentify isomers: 100%|███████████████| 102/102 [00:04<00:00, 21
.88molecule/s]
Analyze atoms: 100%|████████████████| 102/102 [00:00<00:00, 1133
3.25molecule/s]
Collect reaction paths: 100%|██████████████| 150/150 [00:04<00:00,
 33.88atom/s]
Analyze reactions (A+B->C+D): 100%|█| 1000/1000 [00:04<00:00, 214.87timestep/s]

[32m2022-01-08 15:08:13,234[0m - ReacNetGen 1.0.0 - [1;30mINFO:[0m Step 3: D
one! Time consumed (s): 15.995 (Indentify isomers and collect reaction paths)
F:\anaconda3\Scripts\reacnetgenerator-script.py:10: DeprecationWarning: `np.int`
 is a deprecated alias for the builtin `int`. To silence this warning, use `int`
 by itself. Doing this will not modify any behavior and is safe. When replacing
`np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the preci
sion. If you wish to review your current use, check the release note link for ad
ditional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdo
cs/release/1.20.0-notes.html#deprecations
  sys.exit(_commandline())
[32m2022-01-08 15:08:13,520[0m - ReacNetGen 1.0.0 - [1;30mINFO:[0m Step 4: D
one! Time consumed (s): 0.286 (Reaction matrix generation)
[32m2022-01-08 15:08:13,776[0m - ReacNetGen 1.0.0 - [1;30mINFO:[0m Species a
re:
[32m2022-01-08 15:08:13,777[0m - ReacNetGen 1.0.0 - [1;30mINFO:[0m 1 [HH]
[32m2022-01-08 15:08:13,777[0m - ReacNetGen 1.0.0 - [1;30mINFO:[0m 2 [H]C
[32m2022-01-08 15:08:13,778[0m - ReacNetGen 1.0.0 - [1;30mINFO:[0m 3 C#C
[32m2022-01-08 15:08:13,791[0m - ReacNetGen 1.0.0 - [1;30mERROR:[0m [31mErr
or: cannot draw images. Details: random_state_index is incorrect[0m
  File "reacnetgenerator\_draw.py", line 71, in reacnetgenerator._draw._DrawNetw
ork._draw
  File "F:\anaconda3\lib\site-packages\decorator.py", line 231, in fun
    return caller(func, *(extras + args), **kw)
  File "F:\anaconda3\lib\site-packages\networkx\utils\decorators.py", line 400,
in _random_state
    raise nx.NetworkXError("random_state_index is incorrect") from e
[32m2022-01-08 15:08:13,882[0m - ReacNetGen 1.0.0 - [1;30mINFO:[0m Step 5: D
one! Time consumed (s): 0.362 (Draw reaction network)
3it [00:05,  1.81s/it]
Traceback (most recent call last):
  File "F:\anaconda3\Scripts\reacnetgenerator-script.py", line 10, in <module>
    sys.exit(_commandline())
  File "reacnetgenerator\commandline.py", line 59, in reacnetgenerator.commandli
ne._commandline
  File "reacnetgenerator\reacnetgen.py", line 131, in reacnetgenerator.reacnetge
n.ReacNetGenerator.runanddraw
  File "reacnetgenerator\reacnetgen.py", line 188, in reacnetgenerator.reacnetge
n.ReacNetGenerator._process
  File "reacnetgenerator\_reachtml.py", line 45, in reacnetgenerator._reachtml._
HTMLResult.report
  File "reacnetgenerator\_reachtml.py", line 137, in reacnetgenerator._reachtml.
_HTMLResult._generateresult
  File "reacnetgenerator\_reachtml.py", line 157, in reacnetgenerator._reachtml.
_HTMLResult._generatenetwork
FileNotFoundError: [Errno 2] No such file or directory: 'dump.reaxc.svg'

(base) E:\ChromeCoreDownloads>

**Comment by njzjz** [comment]
This is a known issue of NetworkX (https://github.com/networkx/networkx/issues/4718). The issue should have been fixed in a newer version.

**Reply by pig-pig-man2020** [reply]
Installed by "conda install reacnetgenerator -c conda-forge", i got version 1.4.162. Is the newer one 1.4.165?

**Reply by njzjz** [reply]
You need to check the version of networkx.

**Reply by pig-pig-man2020** [reply]
thank you very much
i update NetworkX=2.6.3 and got what i want with the example dump file and a small dump file of mine
but when i run with a larger dump file of mine ( which is just more md steps than the small dump file i mentioned above ) , something wrong  happened 

(base) PS F:\> reacnetgenerator --type lammpsdumpfile -i .\H_O.dump -a H O
[32m2022-01-08 18:19:02,558[0m - ReacNetGen 1.0.0 - [1;30mINFO:[0m ReacNetGe
nerator: an automatic reaction network generator for reactive
molecular dynamics simulation.

Plase cite: ReacNetGenerator: an automatic reaction network generator
for reactive molecular dynamic simulations, Phys. Chem. Chem. Phys.,
2020, 22 (2): 683–691, doi: 10.1039/C9CP05091D

Jinzhe Zeng (jinzhe.zeng@rutgers.edu), Tong Zhu (tzhu@lps.ecnu.edu.cn)

==================
Features
==================
* Processing of MD trajectory containing atomic coordinates or bond orders
* Hidden Markov Model (HMM) based noise filtering
* Isomers identifying accoarding to SMILES
* Generation of reaction network for visualization using force-directed
  algorithm
* Parallel computing

==================
Simple example
==================
ReacNetGenerator can process any kind of trajectory files containing
atomic coordinates, e.g. a LAMMPS dump file prepared by running “dump 1
all custom 100 dump.reaxc id type x y z” in LAMMPS:
$ reacnetgenerator --dump -i dump.reaxc -a C H O
where C, H, and O are atomic names in the input file. Analysis report
will be generated automatically.

Also, ReacNetGenerator can process files containing bond information,
e.g. LAMMPS bond file:
$ reacnetgenerator -i bonds.reaxc -a C H O

You can running the following script for help:
$ reacnetgenerator -h

[32m2022-01-08 18:19:02,560[0m - ReacNetGen 1.0.0 - [1;30mINFO:[0m Version:
1.0.0  Creation date: 2018-03-11  Update date: 2019-10-07
Read bond information and Detect molecules: 101timestep [00:13, 10.23timestep/s]
Read bond information and Detect molecules: 201timestep [00:14, 23.03timestep/s]
Read bond information and Detect molecules: 301timestep [00:15, 37.32timestep/s]
Read bond information and Detect molecules: 401timestep [00:22, 22.62timestep/s]
Read bond information and Detect molecules: 501timestep [00:23, 30.54timestep/s]
Read bond information and Detect molecules: 601timestep [00:23, 45.15timestep/s]
Read bond information and Detect molecules: 701timestep [00:25, 44.91timestep/s]
Read bond information and Detect molecules: 801timestep [00:30, 31.96timestep/s]
Read bond information and Detect molecules: 901timestep [00:32, 37.35timestep/s]
Read bond information and Detect molecules: 1001timestep [00:32, 51.84timestep/s
Read bond information and Detect molecules: 1101timestep [00:35, 47.40timestep/s
Read bond information and Detect molecules: 1201timestep [00:39, 35.28timestep/s
Read bond information and Detect molecules: 1301timestep [00:41, 39.74timestep/s
Read bond information and Detect molecules: 1401timestep [00:43, 40.72timestep/s
Read bond information and Detect molecules: 1501timestep [00:44, 51.10timestep/s
Read bond information and Detect molecules: 1601timestep [00:48, 38.30timestep/s
Read bond information and Detect molecules: 1701timestep [00:52, 35.98timestep/s
Read bond information and Detect molecules: 1801timestep [00:53, 41.04timestep/s
Read bond information and Detect molecules: 1901timestep [00:54, 53.93timestep/s
Read bond information and Detect molecules: 2001timestep [00:57, 42.82timestep/s
Read bond information and Detect molecules: 2101timestep [01:01, 35.47timestep/s
Read bond information and Detect molecules: 2195timestep [01:01, 49.02timestep/s
Read bond information and Detect molecules: 2228timestep [01:02, 45.59timestep/s
Read bond information and Detect molecules: 2301timestep [01:03, 50.87timestep/s
Read bond information and Detect molecules: 2401timestep [01:06, 45.35timestep/s
Read bond information and Detect molecules: 2500timestep [01:07, 36.98timestep/s
]
[32m2022-01-08 18:20:10,388[0m - ReacNetGen 1.0.0 - [1;30mERROR:[0m [31mrun
_mp failed[0m
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "F:\anaconda3\lib\multiprocessing\pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "F:\anaconda3\lib\multiprocessing\pool.py", line 48, in mapstar
    return list(map(*args))
  File "reacnetgenerator\_detect.py", line 281, in reacnetgenerator._detect._Det
ectLAMMPSdump._readstepfunc
UnboundLocalError: local variable 'linecontent' referenced before assignment
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "reacnetgenerator\utils.py", line 214, in reacnetgenerator.utils.run_mp
  File "F:\anaconda3\lib\site-packages\tqdm\std.py", line 1178, in __iter__
    for obj in iterable:
  File "F:\anaconda3\lib\multiprocessing\pool.py", line 448, in <genexpr>
    return (item for chunk in result for item in chunk)
  File "F:\anaconda3\lib\multiprocessing\pool.py", line 868, in next
    raise value
UnboundLocalError: local variable 'linecontent' referenced before assignment
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "F:\anaconda3\lib\multiprocessing\pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "F:\anaconda3\lib\multiprocessing\pool.py", line 48, in mapstar
    return list(map(*args))
  File "reacnetgenerator\_detect.py", line 281, in reacnetgenerator._detect._Det
ectLAMMPSdump._readstepfunc
UnboundLocalError: local variable 'linecontent' referenced before assignment
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "F:\anaconda3\Scripts\reacnetgenerator-script.py", line 10, in <module>
    sys.exit(_commandline())
  File "reacnetgenerator\commandline.py", line 59, in reacnetgenerator.commandli
ne._commandline
  File "reacnetgenerator\reacnetgen.py", line 131, in reacnetgenerator.reacnetge
n.ReacNetGenerator.runanddraw
  File "reacnetgenerator\reacnetgen.py", line 178, in reacnetgenerator.reacnetge
n.ReacNetGenerator._process
  File "reacnetgenerator\_detect.py", line 74, in reacnetgenerator._detect._Dete
ct.detect
  File "reacnetgenerator\_detect.py", line 82, in reacnetgenerator._detect._Dete
ct._readinputfile
  File "reacnetgenerator\_detect.py", line 85, in reacnetgenerator._detect._Dete
ct._readinputfile
  File "reacnetgenerator\utils.py", line 220, in run_mp
  File "reacnetgenerator\utils.py", line 214, in reacnetgenerator.utils.run_mp
  File "F:\anaconda3\lib\site-packages\tqdm\std.py", line 1178, in __iter__
    for obj in iterable:
  File "F:\anaconda3\lib\multiprocessing\pool.py", line 448, in <genexpr>
    return (item for chunk in result for item in chunk)
  File "F:\anaconda3\lib\multiprocessing\pool.py", line 868, in next
    raise value
UnboundLocalError: local variable 'linecontent' referenced before assignment
(base) PS F:\>

**Reply by njzjz** [reply]
Please check if your file is correct and complete, especially the last frame.

**Reply by pig-pig-man2020** [reply]
thank you very much 大神
i checked my largest dump file and found it really lost some lines compare to other file with less md steps
thank you very much again

**Reply by pig-pig-man2020** [reply]
i lost 2 atoms during the laregst md stimulation and add "thermo_modify   lost ignore" to the input file of lammps
so the dump file is "incomplete" 
write down here for anyone who have the same question in the future
